### PR TITLE
State migration

### PR DESCRIPTION
**Pull request by rigalGit**
State migration changes along with directory actions for cloud block addition.

### CODE REVIEW

File: internal/gitops/gitops.go [hunk 1]
```diff
@@ -252,16 +259,18 @@ func PushCommit(repoPath string, remoteName string, branchName string, github_to
...
+		InsecureSkipTLS: true,
+		RemoteName:      remoteName,
 		Auth: &http.BasicAuth{
 			Username: author.Name,
 			Password: authToken,
 		},
 		Force: force,
 	})
 	if err != nil {
+		fmt.Println("\n\n\n\n\n\n\n", githubToken, "\n\n", author.Name, "\n\n\n\n\n\n ")
```
Comment (absolutelightning) [inline internal/gitops/gitops.go:273]: we can get rid of this.

File: internal/gitops/gitops.go [hunk 2]
```diff
@@ -281,16 +290,30 @@ func CreatePullRequest(repoIdentifier, baseBranch, featureBranch, title, body, g
...
 
-	authToken := github_token
+	authToken := githubToken
 	ctx := context.Background()
 	ts := oauth2.StaticTokenSource(
 		&oauth2.Token{AccessToken: authToken},
 	)
-	tc := oauth2.NewClient(ctx, ts)
+
+	//tc := oauth2.NewClient(ctx, ts)
```
Comment (absolutelightning) [inline internal/gitops/gitops.go:299]: we can remove this comment.

File: internal/gitops/gitops.go [hunk 3]
```diff
@@ -281,16 +290,30 @@ func CreatePullRequest(repoIdentifier, baseBranch, featureBranch, title, body, g
...
+	}
+
+	tc := &netHttp.Client{
+		Transport: ts2,
+	}
 
 	client := github.NewClient(tc)
 
-	draft := true
+	draft := false
```
Comment (absolutelightning) [inline internal/gitops/gitops.go:316]: should we take this input from user?

File: internal/gitops/gitops.go [hunk 4]
```diff
@@ -327,9 +350,9 @@ func ListRemote(repoPath string) ([]string, error) {
 }
 
 // GetGitConfig retrieves a global Git configuration value.
-func GlobalGitConfig() GitUserConfig {
+func GlobalGitConfig(repoPath string) GitUserConfig {
 	// Get the global git config file path
-	repo, err := git.PlainOpen(".")
+	repo, err := git.PlainOpen(repoPath)
 	if err != nil {
 		return GitUserConfig{}
```
Comment (absolutelightning) [inline internal/gitops/gitops.go:357]: should we return error here?

File: internal/provider/provider.go [hunk 5]
```diff
@@ -89,7 +89,7 @@ func (p *tfmProvider) Configure(ctx context.Context, req provider.ConfigureReque
 	// Default values to environment variables, but override
 	// with Terraform configuration value if set.
 
-	githubToken := os.Getenv("TFM_GITHUB_TOKEN")
+	githubToken := os.Getenv("GITHUB_TOKEN")
```
Comment (absolutelightning) [inline internal/provider/provider.go:92]: should we move this contant at the top?

File: internal/provider/state_migrate_resource.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	credsFilePath := strings.Join([]string{homeDir, TfcTokenPath}, "/") //nolint:typecheck
```
Comment (absolutelightning) [inline internal/provider/state_migrate_resource.go:206]: should this be filepath.join()

File: internal/gitops/gitops.go (hunk 3 above)
Comment (vignesh-jayabalan) [reply internal/gitops/gitops.go:316]: We can reserve this interaction for later.

**Review by vignesh-jayabalan — APPROVED** [review]

**Pull request merged**